### PR TITLE
Refactor entrypoint

### LINKED ISSUE #6: Don't open port during initialization
When creating the user, we shouldn't open the postgres port.

### PR DESCRIPTION
**Pull request by rhcarvalho**
Unify flows for standalone execution and master of a master-slave setup.

This allows starting with standalone and later adding slaves, as well as remove one extra code path to maintain.

Also fixes #6.

Commits are split logically, but I can rearrange them if needed.

### CODE REVIEW

**Comment by rhcarvalho** [comment]
@bparees @mfojtik @mnagy PTAL.

[test]

File: 9.2/run-postgresql-master.sh [hunk 1]
```diff
@@ -2,27 +2,16 @@
 
 set -eu
 
-source ${HOME}/common.sh
+source $HOME/common.sh
```
Comment (mfojtik) [inline 9.2/run-postgresql-master.sh]: it is a good practice in bash to use "${HOME}"

**Comment by bparees** [comment]
lgtm.

File: 9.2/run-postgresql-master.sh (hunk 1 above)
Comment (rhcarvalho) [inline 9.2/run-postgresql-master.sh]: Hmm, I noticed we use `${HOME}` in some places, `$HOME` in others -- while the curly braces are not needed in that particular case.
What's this good practice about?

**Comment by rhcarvalho** [comment]
[test] again, please

File: 9.2/run-postgresql-master.sh (hunk 1 above)
Comment (mfojtik) [inline 9.2/run-postgresql-master.sh]: https://google-styleguide.googlecode.com/svn/trunk/shell.xml#Variable_expansion

**Comment by rhcarvalho** [comment]
@bparees seems we have some infra problem with tests, failure:

> This system is not registered to Red Hat Subscription Management. You can use subscription-manager to register.

File: 9.2/run-postgresql-master.sh (hunk 1 above)
Comment (rhcarvalho) [inline 9.2/run-postgresql-master.sh]: Thanks for pointing that out. I didn't know we followed anything specific like that.

> In order of precedence: Stay consistent with what you find; ...; prefer "${var}" over "$var"

In multiple places in the same file and group of files we're using a mix of both styles. @mfojtik @bparees what do you think about a follow up making those consistent?!

One way or another is fine for me, as long as we keep ourselves consistent.

Comment (mfojtik) [inline 9.2/run-postgresql-master.sh]: not a high prio... i am fixing those when i touching files :)

**Comment by rhcarvalho** [comment]
[test]

File: 9.2/run-postgresql.sh [hunk 2]
```diff
@@ -16,27 +14,13 @@ else
...
-if [ "$1" = "postgres" ]; then
-  if [ ! -f "$PGDATA/postgresql.conf" ]; then
-    initialize_database
-  fi
-  pg_ctl -w start
-  set_passwords
-  pg_ctl stop
-fi
+case "${1:-}" in
+  'postgres-slave' )
```
Comment (mfojtik) [inline 9.2/run-postgresql.sh]: `postgresql-slave` keep it consistent

File: 9.2/test/run [hunk 3]
```diff
@@ -198,6 +197,12 @@ function run_tests() {
   if [ -v ADMIN_PASS ]; then
     envs="$envs -e POSTGRESQL_ADMIN_PASSWORD=$ADMIN_PASS"
   fi
+  if [ -v POSTGRESQL_MAX_CONNECTIONS ]; then
+    envs="$envs -e POSTGRESQL_MAX_CONNECTIONS=$POSTGRESQL_MAX_CONNECTIONS"
+  fi
+  if [ -v POSTGRESQL_SHARED_BUFFERS ]; then
+    envs="$envs -e POSTGRESQL_SHARED_BUFFERS=$POSTGRESQL_SHARED_BUFFERS"
```
Comment (mfojtik) [inline 9.2/test/run]: envs+=" -e POSTGRESQL_"

File: 9.2/run-postgresql.sh (hunk 2 above)
Comment (rhcarvalho) [inline 9.2/run-postgresql.sh]: not so simple, see https://github.com/openshift/postgresql/pull/51#discussion-diff-38196215L24
It was already spelled that way :/

Comment (mfojtik) [inline 9.2/run-postgresql.sh]: I can't see the original comment :-) Why it is not simple? I guess this is just a helper to start the postgresql image in the "right" mode (iow. a `Cmd` string).

Comment (rhcarvalho) [inline 9.2/run-postgresql.sh]: This is backward incompatible change.
Before `docker run ... postgresql-92-centos7 postgres-slave` (or postgres-master) would work, if we change it, it won't.
Existing deployed templates refer to the old name, if they get a new image is breaks the deployment.

If you don't care about anybody using this image in the wild... then it is indeed "simple": change it here and in the template, and perhaps in documentation if there is any.

**Comment by rhcarvalho** [comment]
please [test] again

**Comment by rhcarvalho** [comment]
and now... [test]

**Comment by mfojtik** [comment]
LGTM [merge]

**Pull request merged**